src/lib/optimizer/strategy/chunk_pruning_rule.cpp

```diff
@@ -184,4 +201,36 @@ bool ChunkPruningRule::_is_non_filtering_node(const AbstractLQPNode& node) const
   return node.type == LQPNodeType::Alias || node.type == LQPNodeType::Projection || node.type == LQPNodeType::Sort;
 }
 
+std::shared_ptr<TableStatistics> ChunkPruningRule::_prune_table_statistics(const TableStatistics& old_statistics,
+                                                                           OperatorScanPredicate predicate,
+                                                                           size_t num_values_pruned) const {
+  // If a chunk is pruned, we update the table statistics. This is so that the selectivity of the predicate that was
+  // used for pruning can be correctly estimated. Example: For a table that has sorted values from 1 to 100 and a chunk
+  // size of 10, the predicate `x > 90` has a selectivity of 10%. However, if the ChunkPruningRule removes nine chunks
+  // out of ten, the selectivity is now 100%. Updating the statistics is important so that the predicate ordering
+  // can properly order the predicates.
+  //
+  // For the column that the predicate pruned on, we remove chunk.size() values that do not match the predicate from
+  // the histogram, assuming equidistribution among the pruned values. The other columns are simply scaled to reflect
```

Comment:
Could you elaborate? How would we do that?

Reply:
Looks like equidistribution is a term. However, I would use uniform since this term is much more widely used (and means the same if I am not mistaken).

Concerning the histogram, I guess @Bensk1  means that two pruned bins with vary different element counts might be adapted differently. Not assuming that every bin "lost" the same number of elements but rather make the loss proportional to the bin size. Is that what you mean, @Bensk1 ?